pkg/upload/path_provider_test.go

```diff
@@ -0,0 +1,106 @@
+package upload
+
+import (
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestPathProvider(t *testing.T) {
```

Comment:
Maybe add a test that uses the defaults as well

Reply:
Didn't want to wait 5min - 1h to testing the default nor creating 50k objects.
I think the system test for the GC can capture this one.